src/charm.py

```diff
@@ -4,25 +4,26 @@
 
 """Charmed operator for the 5G SMF service."""
 
+
 import logging
-from typing import Union
+from ipaddress import IPv4Address
+from subprocess import check_output
+from typing import Dict, Optional, Union
```

Comment:
```suggestion
from typing import Optional, Union
```
We can use `dict` as type hints with recent versions of Python.

Reply:
Changed.